### PR TITLE
rubocop: lib/jekyll/collection.rb

### PR DESCRIPTION
**Pull request by ayastreb**
#4885

### CODE REVIEW

**Comment by ayastreb** [comment]
@parkr any thoughts about this PR? 😄

File: lib/jekyll/collection.rb [hunk 1]
```diff
@@ -57,18 +57,9 @@ def read
...
         next if File.directory?(full_path)
         if Utils.has_yaml_header? full_path
-          doc = Jekyll::Document.new(full_path, { :site => site, :collection => self })
-          doc.read
-          if site.publisher.publish?(doc) || !write?
-            docs << doc
-          else
-            Jekyll.logger.debug "Skipped From Publishing:", doc.relative_path
-          end
+          read_docs(full_path)
```
Comment (parkr) [inline lib/jekyll/collection.rb]: In this case, you're just reading one document. Thus, this should be singular: `read_document(full_path)`.

File: lib/jekyll/collection.rb [hunk 2]
```diff
@@ -57,18 +57,9 @@ def read
...
         else
-          relative_dir = Jekyll.sanitized_path(relative_directory,
-            File.dirname(file_path)).chomp("/.")
-          files << StaticFile.new(site, site.source, relative_dir,
-            File.basename(full_path), self)
+          read_files(file_path, full_path)
```
Comment (parkr) [inline lib/jekyll/collection.rb]: Same here – singular: `read_static_file(file_path, full_path)`.

**Comment by parkr** [comment]
Two naming nits, otherwise, LGTM.

**Comment by parkr** [comment]
@jekyll/core, could one of you take a look at this and give it a LGTM?

File: lib/jekyll/collection.rb [hunk 3]
```diff
@@ -204,5 +195,24 @@ def extract_metadata
         {}
       end
     end
+
+    private
+    def read_docs(full_path)
+      doc = Jekyll::Document.new(full_path, { :site => site, :collection => self })
```
Comment (envygeeks) [inline lib/jekyll/collection.rb]:
```ruby
doc = Document.new(full_path, {
  :site => site, :collection => self
})

doc.read
if site.publisher.publish?(doc) || !write?
```

File: lib/jekyll/collection.rb [hunk 4]
```diff
@@ -204,5 +195,24 @@ def extract_metadata
...
+        docs << doc
+      else
+        Jekyll.logger.debug "Skipped From Publishing:", doc.relative_path
+      end
+    end
+
+    private
+    def read_files(file_path, full_path)
+      relative_dir = Jekyll.sanitized_path(relative_directory,
+                                           File.dirname(file_path)).chomp("/.")
```
Comment (envygeeks) [inline lib/jekyll/collection.rb]: This is very uncleanly and offsets the eyes IMO.  Why not just fold the `()` and put the data between?

**Comment by envygeeks** [comment]
Just a few comments! Thanks.

File: lib/jekyll/collection.rb (hunk 3 above)
Comment (parkr) [inline lib/jekyll/collection.rb]: You don't like the 1-line solution there? I think it reads nicely. We could even remove the curly braces:

```ruby
doc = Document.new(full_path, :site => site, :collection => self)
```

Comment (envygeeks) [inline lib/jekyll/collection.rb]: I do like the version you've given @parkr.

**Comment by parkr** [comment]
LGTM. @envygeeks?

**Comment by envygeeks** [comment]
LGTM.

**Comment by parkr** [comment]
@jekyllbot: merge +dev

**Pull request merged**